clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -599,42 +554,51 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
 
                 if (!parsed.isEmpty()) {
                     log.trace("Adding fetched record for partition {} with offset {} to buffered record list", tp, position);
+                    recordsCount = parsed.size();
+                    parsedRecords = new PartitionRecords<>(fetchOffset, tp, parsed);
                     ConsumerRecord<K, V> record = parsed.get(parsed.size() - 1);
-                    this.records.add(new PartitionRecords<>(fetchOffset, tp, parsed));
                     this.sensors.recordsFetchLag.record(partition.highWatermark - record.offset());
+                    this.sensors.recordTopicFetchMetrics(tp.topic(), bytes, parsed.size());
```

Comment:
Can you please explain the reasoning for only recording topic fetch metrics if `!parsed.isEmpty`?